apps/balancer-tools/src/app/apr/(components)/MultiSelectDropdown.tsx

```diff
@@ -0,0 +1,151 @@
+import { BaseInput } from "#/components/Input";
+
+export const MultiSelectDropdown = ({
+  items,
+  placeholderText,
+  onSelectionItemsChange,
+  initialSelectedItems,
+  ...rest
+}: {
+  items: string[];
+  placeholderText: string;
+  onSelectionItemsChange: (items: string[]) => void;
+  initialSelectedItems: string[];
+}) => {
+  const [selectedItems, setSelectedItems] =
+    useState<string[]>(initialSelectedItems);
+
+  function handleOnKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
+    const inputElement = event.target as HTMLInputElement;
+    if (!inputElement.value.length) {
+      if (event.key === "Backspace") {
+        const copySelectedItems = [...selectedItems];
+        copySelectedItems.pop();
+        setSelectedItems(copySelectedItems);
+        onSelectionItemsChange(copySelectedItems);
+      }
+    }
+  }
+
+  const changeHandler = useCallback(() => {
+    return (
+      selectedItem: string | null,
+      downshift: ControllerStateAndHelpers<string>,
+    ) => {
+      if (!selectedItem) return;
+      const i = selectedItems.findIndex((item) => item === selectedItem);
+      if (i === -1) setSelectedItems([...selectedItems, selectedItem]);
+      onSelectionItemsChange([...selectedItems, selectedItem]);
+      downshift.clearSelection();
+    };
+  }, [selectedItems]);
+
+  const removeSelectedItemByIndex = useCallback(
+    (i: number) => {
+      const temp = [...selectedItems];
+      temp.splice(i, 1);
+      setSelectedItems(temp);
+      onSelectionItemsChange(temp);
+    },
+    [selectedItems],
+  );
+
+  const getFilteredItems = useCallback(
+    (inputValue: string | null) => {
+      return items.filter(
+        (item) =>
+          !selectedItems.find((selected) => selected === item) &&
+          item.includes(String(inputValue)),
+      );
+    },
+    [selectedItems],
+  );
+
+  return (
+    <div className="relative">
+      <Downshift {...rest} onChange={changeHandler()}>
+        {({
+          getLabelProps,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          isOpen,
+          toggleMenu,
+          inputValue,
+        }) => {
+          return (
+            <div>
+              <div className="relative mt-1">
+                <div className="flex items-center bg-blue4 relative w-full cursor-default overflow-hidden rounded-lg text-left shadow-md focus:outline-none focus-within:ring-2 focus-within:ring-blue6 focus-within:ring-opacity/75 focus-within:ring-offset-2 focus-within:ring-offset-blue3 sm:text-sm">
+                  <label {...getLabelProps()} className="flex-1 mx-1">
+                    <MagnifyingGlassIcon />
+                  </label>
+                  <div className="flex gap-2">
+                    {selectedItems.map((value, idx) => {
+                      return (
+                        <Badge key={value + idx} color="blue">
+                          <div
+                            className="flex gap-2 items-center w-max"
```

Comment:
I think either `gap-1` or adding some more horizontal padding will make this look better.

![Uploading Screenshot 2023-08-31 at 11.51.59.png…]()